src/models/login.model.js

```diff
@@ -33,7 +33,7 @@ const login = {
         if (rows.length > 0) {
           const teacher = P.Teacher.dbToTeacher(rows[0]);
           if (bcrypt.compareSync(password, rows[0].passwordteacher)) {
-            const token = this.generateToken(teacher.id);
+            const token = SToken.generateToken(teacher.firstName, teacher.lastName, 'teacher', 'IG');
```

Comment:
Why did you remove the id from the token ? the other attributes don't have to be unique so when getting the user based on the token errors might be made.

Reply:
Because the token from the UM will be exactly with this shape. I changed it to match with the futur connection.